File: WebContent/org/r3/db/bbs/bbsthreadnode/BBSThreadNodeDTO.js
//
// GPA Gruppo Progetti Avanzati s.r.l. - 
// Progetto UNEP 2012 per PosteItaliane S.p.a.
//
//
// TODO: Readme Section.
//



jQuery.r3Utils.namespace("org.r3.db.bbs.bbsthreadnode");

org.r3.db.bbs.bbsthreadnode.BBSThreadNodeDTO = Class.extend(
{
	 init: function(aDataBean)
	 {  
	 	 	 this.dtrowclass   = "grade_bbsthreadnode";
       this.dtokey       = aDataBean.dtokey   || null;
       this.dtodescr     = aDataBean.dtodescr || null;
       
        this.threadnodeid = aDataBean.threadnodeid || null;
        this.site = aDataBean.site || null;
        this.threadid = aDataBean.threadid || null;
        this.nodetype = aDataBean.nodetype || null;
        this.nodetitle = aDataBean.nodetitle || null;
        this.nodetext = aDataBean.nodetext || null;
        this.nodeparentid = aDataBean.nodeparentid || null;
        this.nodepath = aDataBean.nodepath || null;
        this.userid = aDataBean.userid || null;
        this.creationdate = aDataBean.creationdate || null;
        this.recstatus = aDataBean.recstatus || null;
        this.rectimestamp = aDataBean.rectimestamp || null;
 
	 }
    
  ,compareTo: function(anItemId)
   {
   	   if (!this.dtokey)
   	       return -99;
   	       
   	   if (this.dtokey < anItemId)
   	       return -1;
   	   else if (this.dtokey > anItemId)
   	   	   return  1;
   	   	   
   	   return 0;
   }
   
});

File: src/org/r3/db/bbs/bbsthreadnode/BBSThreadNodeDTO.java
//
// GPA Gruppo Progetti Avanzati s.r.l. - 
// Progetto UNEP 2012 per PosteItaliane S.p.a.
//
//
// TODO: Readme Section.
//


package org.r3.db.bbs.bbsthreadnode;

public class BBSThreadNodeDTO extends BBSThreadNodeDTOBase
{	
    public BBSThreadNodeDTO()
    {
        super();
    }

    public BBSThreadNodeDTO(BBSThreadNodeDTO another)
    {
        super(another);
    }

    @com.fasterxml.jackson.annotation.JsonIgnore
    @javax.xml.bind.annotation.XmlTransient    
    public int getNodeDepth()
    {
        if (nodepath == null)
        	return -1;
        
        // Tolgo il punto iniziale e considero l'id da 15 e il punto finale.
        return (nodepath.length() - 1) / 16;
    }
        
    @com.fasterxml.jackson.annotation.JsonIgnore
    @javax.xml.bind.annotation.XmlTransient    
    public boolean isRoot()
    {
    	return nodeparentid == null;
    } 

    @com.fasterxml.jackson.annotation.JsonIgnore
    @javax.xml.bind.annotation.XmlTransient    
    public String getChildrenNodepathPrefix()
    {
    	StringBuilder np = new StringBuilder();
    	np.append(getNodepath()).append(threadnodeid).append('.');

    	return np.toString();
    } 


}
